tests/api2/test_090_boot.py

```diff
@@ -8,51 +8,43 @@
 import os
 from time import time, sleep
 from pytest_dependency import depends
+from middlewared.test.integration.utils import call
 apifolder = os.getcwd()
 sys.path.append(apifolder)
-from functions import GET
 
 
-@pytest.mark.dependency(name='BOOT_DISKS')
+@pytest.mark.dependency(name="BOOT_DISKS")
 def test_01_get_boot_disks():
-    results = GET('/boot/get_disks/')
-    assert results.status_code == 200, results.text
-    disks = results.json()
-    assert isinstance(disks, list) is True, results.text
-    assert disks, results.text
+    call("boot.get_disks")
 
 
-@pytest.mark.dependency(name='BOOT_STATE')
+@pytest.mark.dependency(name="BOOT_STATE")
 def test_02_get_boot_state(request):
-    depends(request, ['BOOT_DISKS'])
-    results = GET('/boot/get_state/')
-    assert results.status_code == 200, results.text
-    assert isinstance(results.json(), dict) is True, results.text
+    depends(request, ["BOOT_DISKS"])
     global boot_state
-    boot_state = results.json()
+    results = call("boot.get_state")
+    boot_state = results
 
 
-@pytest.mark.dependency(name='BOOT_SCRUB')
+@pytest.mark.dependency(name="BOOT_SCRUB")
```

Comment:
Can we maybe opportunistically remove the `depends` usage and any tests that are somewhat pointless.

Reply:
Gotcha
The depends isn't necessary because pytest does the tests in order, correct?

Reply:
It's fixing a long-running anti-pattern in test development that we've been incrementally addressing. Generally setup or shared information between tests should be done through pytest fixtures.